lib/rdoc/parser/prism_ruby.rb

```diff
@@ -0,0 +1,975 @@
+    meth.call_seq = signature
+    return unless meth.name
+
+    meth.start_collecting_tokens
+    node = @line_nodes[line_no]
+    tokens = node ? visible_tokens_from_location(node.location) : [file_line_comment_token(start_line)]
+    tokens.each { |token| meth.token_stream << token }
+
+    container.add_method meth
+    comment.remove_private
+    comment.normalize
+    meth.comment = comment
+    @stats.add_method meth
+  end
+
+  def handle_modifier_directive(code_object, line_no) # :nodoc:
+    comment = @modifier_comments[line_no]
+    @preprocess.handle(comment.text, code_object) if comment
+  end
+
+  def handle_consecutive_comment_directive(code_object, comment) # :nodoc:
+    return unless comment
+    @preprocess.handle(comment, code_object) do |directive, param|
+      case directive
+      when 'method', 'singleton-method',
+           'attr', 'attr_accessor', 'attr_reader', 'attr_writer' then
+        # handled elsewhere
+        ''
+      when 'section' then
+        @container.set_current_section(param, comment.dup)
+        comment.text = ''
+        break
+      end
+    end
+    comment.remove_private
+  end
+
+  def call_node_name_arguments(call_node) # :nodoc:
+    return [] unless call_node.arguments
+    call_node.arguments.arguments.map do |arg|
+      case arg
+      when Prism::SymbolNode
+        arg.value
+      when Prism::StringNode
+        arg.unescaped
+      end
+    end || []
+  end
+
+  # Handles meta method comments
+
+  def handle_meta_method_comment(comment, node)
+    is_call_node = node.is_a?(Prism::CallNode)
+    singleton_method = false
+    visibility = @visibility
+    attributes = rw = line_no = method_name = nil
+
+    processed_comment = comment.dup
+    @preprocess.handle(processed_comment, @container) do |directive, param, line|
+      case directive
+      when 'attr', 'attr_reader', 'attr_writer', 'attr_accessor'
+        attributes = [param] if param
+        attributes ||= call_node_name_arguments(node) if is_call_node
+        rw = directive == 'attr_writer' ? 'W' : directive == 'attr_accessor' ? 'RW' : 'R'
+        ''
+      when 'method'
+        method_name = param
+        line_no = line
+        ''
+      when 'singleton-method'
+        method_name = param
+        line_no = line
+        singleton_method = true
+        visibility = :public
+        ''
+      when 'section' then
+        @container.set_current_section(param, comment.dup)
+        return # If the comment contains :section:, it is not a meta method comment
+      end
+    end
+
+    if attributes
+      attributes.each do |attr|
+        a = RDoc::Attr.new(@container, attr, rw, processed_comment)
+        a.store = @store
+        a.line = line_no
+        a.singleton = @singleton
+        record_location(a)
+        @container.add_attribute(a)
+        a.visibility = visibility
+      end
+    elsif line_no || node
+      method_name ||= call_node_name_arguments(node).first if is_call_node
```

Comment:
Could we reuse `add_method` for this case?

Reply:
Reusing is a bit difficult because of the current comment handling. For now, I extracted common part to `def internal_add_method`.
I think we can make it clean after comment parsing logic is refactored.